test/storage/data_table_test.cpp

```diff
@@ -321,4 +335,33 @@ TEST_F(DataTableTests, InsertNoWrap) {
     delete txn;
   }
 }
+
+// Test insertion into a block, mimicing behaviour while called from the Compactor.
+// A new block is created, then a random tuple is inserted into it.
+// NOLINTNEXTLINE
+TEST_F(DataTableTests, InsertIntoFreezingBlock) {
+  const uint16_t max_columns = 10;
+  RandomDataTableTestObject tested(&block_store_, max_columns, null_ratio_(generator_), &generator_);
+
+  // Create a new block
+  storage::RawBlock *block = tested.GetTable().CreateCompactedBlock();
+
+  // The Compacted Block created should be FREEZING by default
+  EXPECT_EQ(block->controller_.GetBlockState()->load(), storage::BlockState::FREEZING);
+  // Verify that the insertion_head of the block is set at 0 (new block)
+  EXPECT_EQ(block->GetInsertHead(), 0);
+
+  // Insert a random tuple into the newly created block
+  transaction::timestamp_t timestamp(0);
+  auto *txn =
+      new transaction::TransactionContext(timestamp, timestamp, common::ManagedPointer(&buffer_pool_), DISABLED);
+  tested.InsertRandomTupleIntoFreezingBlock(txn, &generator_, block);
+
+  // Once the tuple is inserted, the insertion head should be incremented by 1
+  EXPECT_EQ(block->GetInsertHead(), 1);
+
```

Comment:
Maybe the test can also check the size of inseted_tuples and the actual tuple content being inserted.

Reply:
We decided to move away from this design, marking as resolved.